process/block/preprocess/transactions.go

```diff
@@ -417,11 +434,21 @@ func (txs *transactions) CreateAndProcessMiniBlock(sndShardId, dstShardId uint32
 	log.Info(fmt.Sprintf("creating mini blocks has been started: have %d txs in pool for shard id %d\n", len(orderedTxes), miniBlock.ReceiverShardID))
 
 	addedTxs := 0
+	addedGasLimitPerCrossShardMiniblock := uint64(0)
 	for index := range orderedTxes {
 		if !haveTime() {
 			break
 		}
 
+		currTxGasLimit := minGasLimitForTx
+		if isTxRcvAddressOfSmartContract(orderedTxes[index].RcvAddr) {
+			currTxGasLimit = orderedTxes[index].GasLimit
+
+			if addedGasLimitPerCrossShardMiniblock+currTxGasLimit > process.MaxGasLimitPerMiniBlock {
```

Comment:
Do you want to test that the maxGasLimitPerMiniblock is exceeded just for SC txs? What happens if a if SC is added that uses MaxGasLimitPerMiniBlock - 1 and another 1000 normal txs are added: you would exceed the MaxGasLimitPerMiniblock?

Reply:
ok. condition moved outside. good catch.

Reply:
ok